src/Encoder.h

```diff
@@ -49,13 +45,10 @@ namespace cppkin
     {
     public:
         EncoderImpl();
-        virtual std::string ToString(const Span&) const;
-        virtual std::string ToString(const std::vector<EncoderContext::ContextElement>&) const;
-        virtual std::string ToString(const std::vector<Span*>&) const;
+        std::string ToString(const Span&) const override;
+        std::string ToString(const std::vector<EncoderContext::ContextElement>&) const override;
```

Comment:
Excellent, i really like the fact that we have a more tight interface, the only mean to serialize is by using the context and not the Spans directly, excellent work.